File: 프로그래머스/unrated/134239. 우박수열 정적분/우박수열 정적분.js
function solution(k, ranges) {
    const ks = []
    let nk
    while (k !== 1) {
        nk = k%2===0 ? k/2: k*3+1
        ks.push((k+nk)/2)
        k = nk
    }

    return ranges.map(([s,e]) => 
        s<=ks.length+e
        ? ks.slice(s,ks.length+e).reduce((a,b) => a+b,0)
        : -1
    )
}

File: 프로그래머스/unrated/134239. 우박수열 정적분/우박수열 정적분.py
def solution(k, ranges):
    ks = []
    while k != 1:
        nk = k//2 if k%2==0 else k*3 + 1
        ks.append(sum((k,nk))/2)
        k=nk
    return [
        sum(ks[i:len(ks)+j]) if i<=len(ks)+j else -1
        for i,j in ranges
    ] 
